<a href="https://colab.research.google.com/github/JitendraNathChandra/Data_Science_JNC_Oasis_Infobyte/blob/main/DataScienceJNC_Oasis_CarPrediction.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#  **Name: Jitendra Nath Chandra**
 Registration No: OIB/L2/IP11376

*Oasis Infobyte Internship 2023*

# Task no: 3  CAR PRICE PREDICTION WITH MACHINE LEARNING

**About:** *The price of a car depends on a lot of factors like the goodwill of the brand of the car,features of the car, horsepower and the mileage it gives and many more. Car price prediction is one of the major research areas in machine learning. So if you want to learn how to train a car price prediction model then this project is for you.*

In [4]:
#Importing Required Libraries

import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sb
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.linear_model import Lasso
from sklearn import metrics

Load Data

In [5]:
# loading the data from csv file to pandas dataframe
#Load Data and csv means comma separated values, pd converts text to csv ( separated with comma)

cds = pd.read_csv('CarPrice_Assignment.csv')

Show the data

In [7]:
# inspecting the first 10 rows of the dataframe
#head() means 1st 5 rows of the dataframe

cds.head(10)

car_ID  symboling                   CarName fueltype aspiration doornumber  \
0       1          3        alfa-romero giulia      gas        std        two   
1       2          3       alfa-romero stelvio      gas        std        two   
2       3          1  alfa-romero Quadrifoglio      gas        std        two   
3       4          2               audi 100 ls      gas        std       four   
4       5          2                audi 100ls      gas        std       four   
5       6          2                  audi fox      gas        std        two   
6       7          1                audi 100ls      gas        std       four   
7       8          1                 audi 5000      gas        std       four   
8       9          1                 audi 4000      gas      turbo       four   
9      10          0       audi 5000s (diesel)      gas      turbo        two   

       carbody drivewheel enginelocation  wheelbase  ...  enginesize  \
0  convertible        rwd          front       88.6  ...         130   
1  convertible        rwd          front       88.6  ...         130   
2    hatchback        rwd          front       94.5  ...         152   
3        sedan        fwd          front       99.8  ...         109   
4        sedan        4wd          front       99.4  ...         136   
5        sedan        fwd          front       99.8  ...         136   
6        sedan        fwd          front      105.8  ...         136   
7        wagon        fwd          front      105.8  ...         136   
8        sedan        fwd          front      105.8  ...         131   
9    hatchback        4wd          front       99.5  ...         131   

   fuelsystem  boreratio  stroke compressionratio horsepower  peakrpm citympg  \
0        mpfi       3.47    2.68              9.0        111     5000      21   
1        mpfi       3.47    2.68              9.0        111     5000      21   
2        mpfi       2.68    3.47              9.0        154     5000      19   
3        mpfi       3.19    3.40             10.0        102     5500      24   
4        mpfi       3.19    3.40              8.0        115     5500      18   
5        mpfi       3.19    3.40              8.5        110     5500      19   
6        mpfi       3.19    3.40              8.5        110     5500      19   
7        mpfi       3.19    3.40              8.5        110     5500      19   
8        mpfi       3.13    3.40              8.3        140     5500      17   
9        mpfi       3.13    3.40              7.0        160     5500      16   

   highwaympg      price  
0          27  13495.000  
1          27  16500.000  
2          26  16500.000  
3          30  13950.000  
4          22  17450.000  
5          25  15250.000  
6          25  17710.000  
7          25  18920.000  
8          20  23875.000  
9          22  17859.167  

[10 rows x 26 columns]

In [9]:
#if you provide n numbers rows of data then head() <-- 'n' put it
cds.head(50)

car_ID  symboling                    CarName fueltype aspiration  \
0        1          3         alfa-romero giulia      gas        std   
1        2          3        alfa-romero stelvio      gas        std   
2        3          1   alfa-romero Quadrifoglio      gas        std   
3        4          2                audi 100 ls      gas        std   
4        5          2                 audi 100ls      gas        std   
5        6          2                   audi fox      gas        std   
6        7          1                 audi 100ls      gas        std   
7        8          1                  audi 5000      gas        std   
8        9          1                  audi 4000      gas      turbo   
9       10          0        audi 5000s (diesel)      gas      turbo   
10      11          2                   bmw 320i      gas        std   
11      12          0                   bmw 320i      gas        std   
12      13          0                     bmw x1      gas        std   
13      14          0                     bmw x3      gas        std   
14      15          1                     bmw z4      gas        std   
15      16          0                     bmw x4      gas        std   
16      17          0                     bmw x5      gas        std   
17      18          0                     bmw x3      gas        std   
18      19          2           chevrolet impala      gas        std   
19      20          1      chevrolet monte carlo      gas        std   
20      21          0        chevrolet vega 2300      gas        std   
21      22          1              dodge rampage      gas        std   
22      23          1        dodge challenger se      gas        std   
23      24          1                 dodge d200      gas      turbo   
24      25          1          dodge monaco (sw)      gas        std   
25      26          1         dodge colt hardtop      gas        std   
26      27          1            dodge colt (sw)      gas        std   
27      28          1       dodge coronet custom      gas      turbo   
28      29         -1          dodge dart custom      gas        std   
29      30          3  dodge coronet custom (sw)      gas      turbo   
30      31          2                honda civic      gas        std   
31      32          2           honda civic cvcc      gas        std   
32      33          1                honda civic      gas        std   
33      34          1          honda accord cvcc      gas        std   
34      35          1           honda civic cvcc      gas        std   
35      36          0            honda accord lx      gas        std   
36      37          0        honda civic 1500 gl      gas        std   
37      38          0               honda accord      gas        std   
38      39          0           honda civic 1300      gas        std   
39      40          0              honda prelude      gas        std   
40      41          0               honda accord      gas        std   
41      42          0                honda civic      gas        std   
42      43          1         honda civic (auto)      gas        std   
43      44          0                 isuzu MU-X      gas        std   
44      45          1               isuzu D-Max       gas        std   
45      46          0        isuzu D-Max V-Cross      gas        std   
46      47          2               isuzu D-Max       gas        std   
47      48          0                  jaguar xj      gas        std   
48      49          0                  jaguar xf      gas        std   
49      50          0                  jaguar xk      gas        std   

   doornumber      carbody drivewheel enginelocation  wheelbase  ...  \
0         two  convertible        rwd          front       88.6  ...   
1         two  convertible        rwd          front       88.6  ...   
2         two    hatchback        rwd          front       94.5  ...   
3        four        sedan        fwd      

# checking the number of rows and columns

In [11]:
#Shape --> total number of Rows and Columns

print(cds.shape)


(205, 26)


In this dataframe there are 205 rows and 26 columns

## Describe method: Statistic of our Data ( like- Count, Mean, Std, Min, Max)

In [13]:
cds.describe()

car_ID   symboling   wheelbase   carlength    carwidth   carheight  \
count  205.000000  205.000000  205.000000  205.000000  205.000000  205.000000   
mean   103.000000    0.834146   98.756585  174.049268   65.907805   53.724878   
std     59.322565    1.245307    6.021776   12.337289    2.145204    2.443522   
min      1.000000   -2.000000   86.600000  141.100000   60.300000   47.800000   
25%     52.000000    0.000000   94.500000  166.300000   64.100000   52.000000   
50%    103.000000    1.000000   97.000000  173.200000   65.500000   54.100000   
75%    154.000000    2.000000  102.400000  183.100000   66.900000   55.500000   
max    205.000000    3.000000  120.900000  208.100000   72.300000   59.800000   

        curbweight  enginesize   boreratio      stroke  compressionratio  \
count   205.000000  205.000000  205.000000  205.000000        205.000000   
mean   2555.565854  126.907317    3.329756    3.255415         10.142537   
std     520.680204   41.642693    0.270844    0.313597          3.972040   
min    1488.000000   61.000000    2.540000    2.070000          7.000000   
25%    2145.000000   97.000000    3.150000    3.110000          8.600000   
50%    2414.000000  120.000000    3.310000    3.290000          9.000000   
75%    2935.000000  141.000000    3.580000    3.410000          9.400000   
max    4066.000000  326.000000    3.940000    4.170000         23.000000   

       horsepower      peakrpm     citympg  highwaympg         price  
count  205.000000   205.000000  205.000000  205.000000    205.000000  
mean   104.117073  5125.121951   25.219512   30.751220  13276.710571  
std     39.544167   476.985643    6.542142    6.886443   7988.852332  
min     48.000000  4150.000000   13.000000   16.000000   5118.000000  
25%     70.000000  4800.000000   19.000000   25.000000   7788.000000  
50%     95.000000  5200.000000   24.000000   30.000000  10295.000000  
75%    116.000000  5500.000000   30.000000   34.000000  16503.000000  
max    288.000000  6600.000000   49.000000   54.000000  45400.000000

#Some more information about the Dataset

In [15]:
cds.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 205 entries, 0 to 204
Data columns (total 26 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   car_ID            205 non-null    int64  
 1   symboling         205 non-null    int64  
 2   CarName           205 non-null    object 
 3   fueltype          205 non-null    object 
 4   aspiration        205 non-null    object 
 5   doornumber        205 non-null    object 
 6   carbody           205 non-null    object 
 7   drivewheel        205 non-null    object 
 8   enginelocation    205 non-null    object 
 9   wheelbase         205 non-null    float64
 10  carlength         205 non-null    float64
 11  carwidth          205 non-null    float64
 12  carheight         205 non-null    float64
 13  curbweight        205 non-null    int64  
 14  enginetype        205 non-null    object 
 15  cylindernumber    205 non-null    object 
 16  enginesize        205 non-null    int64  
 1

# checking the number of missing values in each columns


In [17]:
#Given Dataframe contain some null values , we have to remove then from cds
cds.isnull().sum()

car_ID              0
symboling           0
CarName             0
fueltype            0
aspiration          0
doornumber          0
carbody             0
drivewheel          0
enginelocation      0
wheelbase           0
carlength           0
carwidth            0
carheight           0
curbweight          0
enginetype          0
cylindernumber      0
enginesize          0
fuelsystem          0
boreratio           0
stroke              0
compressionratio    0
horsepower          0
peakrpm             0
citympg             0
highwaympg          0
price               0
dtype: int64

# Checking the distribution of categorical data

In [25]:
print(cds.fueltype.value_counts())
print(cds.enginetype.value_counts())
print(cds.carbody.value_counts())
print(cds.enginelocation.value_counts())
print(cds.fuelsystem.value_counts())

gas       185
diesel     20
Name: fueltype, dtype: int64
ohc      148
ohcf      15
ohcv      13
dohc      12
l         12
rotor      4
dohcv      1
Name: enginetype, dtype: int64
sedan          96
hatchback      70
wagon          25
hardtop         8
convertible     6
Name: carbody, dtype: int64
front    202
rear       3
Name: enginelocation, dtype: int64
mpfi    94
2bbl    66
idi     20
1bbl    11
spdi     9
4bbl     3
mfi      1
spfi     1
Name: fuelsystem, dtype: int64


In this dataset we include 185 Gas vehicle and 20 diesel vehicle